x-pack/plugins/security_solution/server/lib/detection_engine/signals/executors/threshold.ts

```diff
@@ -86,8 +87,14 @@ export const thresholdExecutor = async ({
       result.warningMessages.push(exceptionsWarning);
     }
 
+    // Invalidate state history if rule's essantial properties were changed
+    const ruleEssentialsHash = getRuleEssentialsHash(completeRule);
+    const useSignalHistory =
+      state.initialized &&
+      (state.ruleEssentialsHash == null || state.ruleEssentialsHash === ruleEssentialsHash);
+
     // Get state or build initial state (on upgrade)
-    const { signalHistory, searchErrors: previousSearchErrors } = state.initialized
+    const { signalHistory, searchErrors: previousSearchErrors } = useSignalHistory
```

Comment:
If I understand correctly, when `ruleEssentialsHash` is not equal to one in store, we will call `getThresholdSignalHistory`

```ts
await getThresholdSignalHistory({
          from: tuple.from.toISOString(),
          to: tuple.to.toISOString(),
          frameworkRuleId: completeRule.alertId,
          bucketByFields: ruleParams.threshold.field,
          ruleDataReader,
        });

```

And this function will create new history record based only on range, and `ruleParams.threshold.field`.
So, if in essential rule parameter would change, query for example, it wouldn't be take into account when new history will be created and used to build ES filter. 

Should we at all create new history in that case if essential params changed?